tools/RepositoryManagement/DevHome.RepositoryManagement/Services/RepositoryManagementDataAccessService.cs

```diff
@@ -0,0 +1,104 @@
+using Microsoft.EntityFrameworkCore;
+using Microsoft.EntityFrameworkCore.Query;
+using Microsoft.Extensions.Hosting;
+using Serilog;
+
+namespace DevHome.RepositoryManagement.Services;
+
+public class RepositoryManagementDataAccessService
+{
+    private readonly ILogger _log = Log.ForContext("SourceContext", nameof(RepositoryManagementDataAccessService));
+
+    private readonly IHost _host;
+
+    public RepositoryManagementDataAccessService(IHost host)
+    {
+        // Store the host to make dbContext and RepositoryManagementItemViewModel
+        // dbContext can not be passed in because RepositoryManagementDataAccessService is singleton
+        // and dbContext needs to be scoped.
+        // RepositoryManagementItemViewModel can not be passed in because multiple are made.
+        // The best solution is to make factories for DevHomeContext and RepositoryManagementItemViewModel.
+        // Might be in a future change.
+        _host = host;
+    }
+
+    public void AddRepository(string repositoryName, string cloneLocation)
+    {
+        Repository newRepo = new();
+        newRepo.RepositoryName = repositoryName;
+        newRepo.RepositoryClonePath = cloneLocation;
+
+        RepositoryMetadata newMetadata = new();
+        newMetadata.Repository = newRepo;
+        newMetadata.RepositoryId = newRepo.RepositoryId;
+        newMetadata.IsHiddenFromPage = false;
+
+        newRepo.RepositoryMetadata = newMetadata;
+
+        var dbContext = _host.GetService<DevHomeDatabaseContext>();
+        dbContext.Add(newRepo);
+        dbContext.Add(newMetadata);
+        dbContext.SaveChanges();
+    }
+
+    public List<RepositoryManagementItemViewModel> GetRepositories(bool removeRepositoriesNotInTheirSavedLocation)
+    {
+        var repos = QueryDatabaseForRepositories(removeRepositoriesNotInTheirSavedLocation);
+        return ConvertToLineItems(repos);
+    }
+
+    private IIncludableQueryable<Repository, RepositoryMetadata> QueryDatabaseForRepositories(bool removeRepositoriesNotInTheirSavedLocation)
+    {
+        var dbContext = _host.GetService<DevHomeDatabaseContext>();
+        var repositoriesFromDatabase = dbContext.Repositories.Include(x => x.RepositoryMetadata);
+
+        if (!removeRepositoriesNotInTheirSavedLocation)
+        {
+            return repositoriesFromDatabase;
+        }
+
+        // The database can get out of sync.  Take this as an example.
+        // DevHome writes a repository.
+        // The user moves or deletes the repository.
+        // Now the database record has the correct name, but the incorrect location.
+        // Remove records where the directory does not exist.
+        repositoriesFromDatabase
+            .ToList() // EF does not understand Directory.Exists.  Fetch the data first.
+            .Where(x => !Directory.Exists(x.RepositoryClonePath))
+            .ToList()
+            .ForEach(x => dbContext.Repositories.Remove(x));
+
+        dbContext.SaveChanges();
```

Comment:
Not familiar with what magic EntityFramework might be doing, but SQLite has transaction support where you can ensure data integrity should something go wrong with any of this.

I don't see much in the way of try-catch blocks for things going wrong and I'm particularly nervous about not having fine control over the schema, transactions, etc to ensure data integrity.

Reply:
>  but SQLite has transaction support 
By default Entity Framework uses [Transactions](https://learn.microsoft.com/ef/core/saving/transactions#default-transaction-behavior) any `SaveChange`

> try-catch blocks for things going wrong is put inside a transaction.
They will be added later.  The purpose of this PR is to get the database structure and some usage examples into the feature branch.  More try/catches will be added after this.

Reply:
> fine control over the schema, transactions, etc to ensure data integrity.

The schema is defined by the C# classes.  We will have finer control over the schema.  I understand I did not use this in this PR, but I will in the next one.  EF generates [SQL scripts](https://learn.microsoft.com/ef/core/managing-schemas/migrations/applying?tabs=vs#basic-usage-1) from migrations.  When the schema is updated human eyes will review the generated script.

As I said in another comment, EF uses transactions by default.

What other concerns around data integrity do you have?